### PR TITLE
Update upstream to mariadb 10.2.14

### PR DESCRIPTION
**Pull request by fritzm**

### CODE REVIEW

File: ups/eupspkg.cfg.sh [hunk 1]
```diff
@@ -13,7 +13,10 @@ config()
     # TokuDB requires third-party software jemalloci (see http://www.canonware.com/jemalloc/)
     ARGS+=('-DPLUGIN_TOKUDB=NO')
 
-    # Always use bundled PCRE
+    # Due to cmake library discovery being overly energetic (searching every
+    # lib directory relative to each entry in $PATH) we use the bundled ZLIB and
+    # PCRE libraries to avoid link confusion downstream
+    ARGS+=('-DWITH_ZLIB=bundled')
```
Comment (timj) [inline ups/eupspkg.cfg.sh:19]: The commit message isn't really clear about this ZLIB change.

Comment (fritzm) [reply ups/eupspkg.cfg.sh:19]: Oh.  It was part of "Streamline eupspkg.cfg to be more similar with mariadbclient version", but it certainly wouldn't hurt to be more explicit; will fix.

Still losing on OSX, BTW :-/  For some reason, -DWITH_SSL=off doesn't seem to bypass the failing GnuTLS check, though it seems by my first reading of the cmake file that it should.  Installing GnuTLS via homebrew "fixes" it, but I don't think we want to add another prereq if we can avoid it.  I'll continue to fiddle with it...

File: ups/eupspkg.cfg.sh [hunk 2]
```diff
@@ -13,35 +13,17 @@ config()
...
-    # Always use bundled PCRE
+    # Due to cmake library discovery being overly energetic (searching every
+    # lib directory relative to each entry in $PATH) we use the bundled SSL and
+    # ZLIB libraries to avoid link confusion downstream
+    ARGS+=('-DWITH_ZLIB=bundled')
     ARGS+=('-DWITH_PCRE=bundled')
 
     # Prevent CMake from finding and linking against libraries distributed in $(dirname
-    # python)/../lib. This is CMake's default behaviour, but can cause us to erroneously link
+    # python)/../lib. This is CMake's default behavior, but can cause us to erroneously link
```
Comment (timj) [inline ups/eupspkg.cfg.sh:23]: Really? English spelling is not acceptable to you? 😛

File: ups/eupspkg.cfg.sh [hunk 3]
```diff
@@ -13,35 +13,17 @@ config()
     # TokuDB requires third-party software jemalloci (see http://www.canonware.com/jemalloc/)
     ARGS+=('-DPLUGIN_TOKUDB=NO')
 
-    # Always use bundled PCRE
+    # Due to cmake library discovery being overly energetic (searching every
+    # lib directory relative to each entry in $PATH) we use the bundled SSL and
```
Comment (timj) [inline ups/eupspkg.cfg.sh]: This mention of SSL is no longer correct, and the mention of PCRE was removed from the comment but it's still here.

**Review by timj — APPROVED** [review]

File: ups/eupspkg.cfg.sh [hunk 4]
```diff
@@ -13,35 +13,17 @@ config()
...
+    # ZLIB libraries to avoid link confusion downstream
+    ARGS+=('-DWITH_ZLIB=bundled')
     ARGS+=('-DWITH_PCRE=bundled')
 
     # Prevent CMake from finding and linking against libraries distributed in $(dirname
-    # python)/../lib. This is CMake's default behaviour, but can cause us to erroneously link
+    # python)/../lib. This is CMake's default behavior, but can cause us to erroneously link
     # against libraries distributed by e.g. Anaconda.
     PYTHONLIBDIR=$(which python | sed -e's|bin/python|lib|')
     ARGS+=("-DCMAKE_SYSTEM_IGNORE_PATH=${PYTHONLIBDIR}")
```
Comment (jhoblitt) [inline ups/eupspkg.cfg.sh:26]: I have the same question here as on https://github.com/lsst/mariadbclient/pull/12

**Pull request merged**